File: HackerRank/Algorithms/1. Warmup/Birthday_cake_candles.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    //Code stub provided
    int n, height_i;
    scanf("%d",&n);
    int *height = malloc(sizeof(int) * n);
    for(height_i = 0; height_i < n; height_i++){
       scanf("%d",&height[height_i]);
    }

    //Code starts
    int max = height[0];
    int count = 1;
    for (height_i = 1; height_i < n; height_i++) {
      if (height[height_i] == max) {
        ++count;
      }
      else if (height[height_i] > max) {
        max = height[height_i];
        count = 1;
      }
    }

    printf("%d\n", count);
    return 0;
}

File: HackerRank/Algorithms/1. Warmup/Birthday_cake_candles.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int height[] = new int[n];
        for(int height_i=0; height_i < n; height_i++){
            height[height_i] = in.nextInt();
        }

        int max = height[0];
        int count = 1;
        for (int i = 1; i < n; ++i) {
          if (height[i] == max) {
            ++count;
          }
          else if (height[i] > max) {
            max = height[i];
            count = 1;
          }
        }

        System.out.println(count);
    }
}
